test/src/tests-batchUploader.ts

```diff
@@ -415,6 +442,704 @@ describe('batch uploader', () => {
+
+            const batchValidator = new _BatchValidator();
+
+            // Create sample batches for testing
+            const batch1 = batchValidator.returnBatch([
+                {
+                    messageType: 4,
+                    name: 'Test Event 1',
+                },
+                {
+                    messageType: 4,
+                    name: 'Test Event 2',
+                },
+            ]);
+
+            const batch2 = batchValidator.returnBatch([
+                {
+                    messageType: 4,
+                    name: 'Test Event 3',
+                },
+                {
+                    messageType: 4,
+                    name: 'Test Event 4',
+                },
+            ]);
+
+            const batch3 = batchValidator.returnBatch([
+                {
+                    messageType: 4,
+                    name: 'Test Event 5',
+                },
+                {
+                    messageType: 4,
+                    name: 'Test Event 6',
+                },
+            ]);
+
+            // Write batches to Offline Storage before queuing new events or batches
+            window.localStorage.setItem(
+                batchStorageKey,
+                JSON.stringify([batch1, batch2, batch3])
+            );
+
+            // Batch Queue should be empty before we upload
+            expect(uploader.batchesQueuedForProcessing.length).to.equal(0);
+
+            // Manually initiate the upload process - turn event into batches and upload the batch
+            window.mParticle.upload();
+
+            clock.restore();
+
+            setTimeout(() => {
+                expect(
+                    window.localStorage.getItem(batchStorageKey),
+                    'Offline Batch Storage should be empty'
+                ).to.equal('');
+
+                // To verify the sequence, we should look at what has been uploaded
+                // as the upload queue and Offline Storage should be empty
+                expect(window.fetchMock.calls().length).to.equal(4);
+
+                const uploadedBatch1: Batch = JSON.parse(
+                    window.fetchMock.calls()[0][1].body
+                );
+                const uploadedBatch2: Batch = JSON.parse(
+                    window.fetchMock.calls()[1][1].body
+                );
+                const uploadedBatch3: Batch = JSON.parse(
+                    window.fetchMock.calls()[2][1].body
+                );
+                const uploadedBatch4: Batch = JSON.parse(
+                    window.fetchMock.calls()[3][1].body
+                );
+
+                // The following assertions should verify the sequence presented below
+                // - Batch 1: Test Event 1 and 2 - From Offline Storage
+                // - Batch 2: Test Event 3 and 4 - From Offline Storage
+                // - Batch 3: Test Event 5 and 6 - From Offline Storage
+                // - Batch 4: Session Start and AST - (new) Created by Init
+
```

Comment:
I think this comment needs to be updated since the comment from 1069-1070 says there isn't anything in offline storage.